src/main/java/com/synopsys/integration/alert/workflow/startup/component/AuthenticationSettingsMigration.java

```diff
@@ -0,0 +1,161 @@
+@Order(2)
+// TODO Remove this class in 6.0.0
+// TODO Revisit the order of the startup components
+public class AuthenticationSettingsMigration extends StartupComponent {
+    private Logger logger = LoggerFactory.getLogger(AuthenticationSettingsMigration.class);
+    private EnvironmentVariableUtility environmentUtility;
+    private DescriptorAccessor descriptorAccessor;
+    private ConfigurationAccessor fieldConfigurationAccessor;
+    private ConfigurationFieldModelConverter modelConverter;
+    private FieldModelProcessor fieldModelProcessor;
+    private AuthenticationDescriptorKey authenticationDescriptorKey;
+    private SettingsUtility settingsUtility;
+
+    @Autowired
+    public AuthenticationSettingsMigration(EnvironmentVariableUtility environmentUtility, DescriptorAccessor descriptorAccessor, ConfigurationAccessor fieldConfigurationAccessor,
+        ConfigurationFieldModelConverter modelConverter, FieldModelProcessor fieldModelProcessor, AuthenticationDescriptorKey authenticationDescriptorKey, SettingsUtility settingsUtility) {
+        this.environmentUtility = environmentUtility;
+        this.descriptorAccessor = descriptorAccessor;
+        this.fieldConfigurationAccessor = fieldConfigurationAccessor;
+        this.modelConverter = modelConverter;
+        this.fieldModelProcessor = fieldModelProcessor;
+        this.authenticationDescriptorKey = authenticationDescriptorKey;
+        this.settingsUtility = settingsUtility;
+    }
+
+    @Override
+    protected void initialize() {
+        logger.info("Settings authentication variable migration start...");
+        try {
+            List<DefinedFieldModel> fieldsForDescriptor = descriptorAccessor.getFieldsForDescriptor(authenticationDescriptorKey, ConfigContextEnum.GLOBAL).stream()
+                                                              .sorted(Comparator.comparing(DefinedFieldModel::getKey))
+                                                              .collect(Collectors.toList());
+            List<ConfigurationModel> foundConfigurationModels = fieldConfigurationAccessor.getConfigurationByDescriptorKeyAndContext(authenticationDescriptorKey, ConfigContextEnum.GLOBAL);
+
+            Map<String, ConfigurationFieldModel> existingConfiguredFields = new HashMap<>();
+            foundConfigurationModels.forEach(config -> existingConfiguredFields.putAll(config.getCopyOfKeyToFieldMap()));
+
+            Set<ConfigurationFieldModel> configurationModels = createFieldModelsFromDefinedFields(fieldsForDescriptor, existingConfiguredFields);
+            boolean overwriteConfig = settingsUtility.getConfiguration()
+                                          .flatMap(configurationModel -> configurationModel.getField(SettingsDescriptor.KEY_STARTUP_ENVIRONMENT_VARIABLE_OVERRIDE))
+                                          .flatMap(ConfigurationFieldModel::getFieldValue)
+                                          .map(value -> Boolean.valueOf(value)).orElse(Boolean.FALSE);
```

Comment:
This can be replaced with a method reference. Also, I generally prefer the orElse on it's own line.